gprofiler/perf.py

```diff
@@ -55,29 +55,23 @@ def _run_perf(self, filename_base: str, dwarf=False):
                 [resource_path("perf")] + buildid_args + ["record"] + args + ["--", "sleep", str(self._duration)],
                 stop_event=self._stop_event,
             )
-            with open(parsed_path, "w") as f:
-                run_process(
-                    [resource_path("perf")] + buildid_args + ["script", "-F", "+pid", "-i", record_file.name], stdout=f
-                )
-            return parsed_path
+            perf_script_result = run_process(
+                [resource_path("perf")] + buildid_args + ["script", "-F", "+pid", "-i", record_file.name]
+            )
+            return perf_script_result.stdout.decode('utf8')
```

Comment:
In that case we're limited to the file buffer size, I think (since you're not reading along the way, only when the process is done). Default size is 65k which is too low for common use-cases. Keep it as a file.

Reply:
We are using `process.communicate()`, it reads along the way